src/java.base/share/classes/java/lang/reflect/code/type/impl/FieldRefImpl.java

```diff
@@ -104,7 +104,7 @@ public VarHandle resolveToHandle(MethodHandles.Lookup l) throws ReflectiveOperat
 
     static Class<?> resolve(MethodHandles.Lookup l, TypeElement t) throws ReflectiveOperationException {
         if (t instanceof JavaType jt) {
-            return jt.resolve(l);
+            return (Class<?>)jt.resolve(l);
```

Comment:
```suggestion
            return (Class<?>)jt.erasure().resolve(l);
```
?

Reply:
I will accept these suggestions. However, note that the code was left unchanged deliberately: I believe when we create method refs and field refs we use always an erased owner type. That said if we wanted to be stricter about this, perhaps we should consider swapping `JavaType` for `ClassDesc` for the owner, or checking the validity of the JavaType at construction.